### PR TITLE
Feature/coa control

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Control whether a CoA or RADIUS disconnect should be used when changing a role
# Impacts

Aruba + Avaya network modules
# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Added control of CoA vs Disconnect when deauthenticating a device.

### CODE REVIEW

**Comment by julsemaan** [comment]
Please review

File: lib/pf/Switch/Aruba.pm [hunk 1]
```diff
@@ -498,7 +498,7 @@ sub radiusDisconnect {
         $attributes_ref = { %$attributes_ref, %$add_attributes_ref };
 
         # Roles are configured and the user should have one
-        if ( defined($role) && (defined($node_info->{'status'}) && isenabled($self->{_RoleMap}) ) ) {
+        if ( defined($role) && isenabled($self->{_RoleMap}) && isenabled($self->{_useCoA}) ) {
```
Comment (jrouzierinverse) [inline lib/pf/Switch/Aruba.pm]: This would be best wrapped in a function.

Comment (julsemaan) [inline lib/pf/Switch/Aruba.pm]: Here we say vrapped.

But I can still agree with you

**Comment by julsemaan** [comment]
Re-review

File: lib/pf/Switch/Aruba.pm (hunk 1 above)
Comment (jrouzierinverse) [inline lib/pf/Switch/Aruba.pm]: vrapped?

Comment (julsemaan) [inline lib/pf/Switch/Aruba.pm]: Forget about it :) just another joke out of my bad jokes bucket (which happen to contain all my jokes)

File: lib/pf/Switch/Avaya.pm [hunk 2]
```diff
@@ -589,8 +589,7 @@ sub radiusDisconnect {
         # merging additional attributes provided by caller to the standard attributes
         $attributes_ref = { %$attributes_ref, %$add_attributes_ref };
 
-        # Roles are configured and the user should have one
-        if (defined($role) && (defined($node_info->{'status'}) ) ) {
+        if ( $self->shouldUseCoA({role => $role}) ) {
```
Comment (jrouzierinverse) [inline lib/pf/Switch/Avaya.pm:592]: This did not previously check for _RoleMap.
Was this a bug in the old code?

Comment (julsemaan) [inline lib/pf/Switch/Avaya.pm:592]: That is correct. It was likely to have been forgotten when RoleMap was brought in.
It should use the same logic as the other modules

**Pull request merged**